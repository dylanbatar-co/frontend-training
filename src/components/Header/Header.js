import React from 'react';
import Navbar from 'emerald-ui/lib/Navbar';
import Nav from 'emerald-ui/lib/Nav';
import DropdownButton from 'emerald-ui/lib/DropdownButton';
import DropdownItem from 'emerald-ui/lib/DropdownItem';
import Avatar from 'emerald-ui/lib/Avatar';

export const Header = () => {
  const dropdownItems = [
    'Sport',
    'Entertainment',
    'Economics',
    'Technology',
    'Games'
  ];

  return (
    <Navbar breakAt="lg" fixedAtTop>
      <Navbar.Brand>
        <a href="/">
          <img
            src={`${process.env.PUBLIC_URL}/assets/img/logo.svg`}
            alt="logo"
          />
        </a>
      </Navbar.Brand>
      <Nav grow collapsible>
        <RenderDropdownList listItem={dropdownItems} titleList="Section" />
        <a href="#editorial">Editorial</a>
        <a href="#contactus">Contact Us</a>
      </Nav>
      <Nav>
        <a href="#avatar">
          <Avatar title="JS" />
        </a>
      </Nav>
    </Navbar>
  );
};

const RenderDropdownList = ({ listItem = [], titleList = '' }) => {
  return (
    <DropdownButton title={titleList} id="dd1">
      {listItem.map((item, idx) => (
        <DropdownItem key={idx} eventKey={idx}>
          {item}
        </DropdownItem>
      ))}
    </DropdownButton>
  );
};
