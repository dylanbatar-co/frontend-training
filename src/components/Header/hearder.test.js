import { shallow } from 'enzyme';
import { Header } from './Header';

const component = shallow(<Header />);

describe('Navbar Tests', () => {
  it('Snapshot navbar', () => {
    expect(component).toMatchSnapshot();
  });

  it('Navbar is fixed-at-top', () => {
    expect(component.getElement().props.fixedAtTop).toBe(true);
  });
});

describe('DropDown Tests', () => {
  it('Should have at least 5 items', () => {
    const NavContainer = component.find('RenderDropdownList');
    const listItems = NavContainer.props().listItem.length;
    expect(listItems).toBeGreaterThanOrEqual(5);
  });

  it('DropDownItem text should be same dropdownItems ', () => {
    const dropdownItems = [
      'Sport',
      'Entertainment',
      'Economics',
      'Technology',
      'Games'
    ];

    const DropdownItem = component
      .find('RenderDropdownList')
      .dive()
      .find('DropdownItem')
      .children()
      .map(el => el.text());

    DropdownItem.forEach((v, i) => expect(v).toEqual(dropdownItems[i]));
  });
});
