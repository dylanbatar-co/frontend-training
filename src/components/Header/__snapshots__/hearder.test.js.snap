// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Tests Snapshot navbar 1`] = `
<Navbar
  breakAt="lg"
  fixedAtTop={true}
  theme="light"
>
  <NavbarBrand
    displayName="NavbarBrand"
  >
    <img
      alt="logo"
      src="/assets/img/logo.svg"
    />
  </NavbarBrand>
  <Nav
    collapsible={true}
    displayName="Nav"
    grow={true}
  >
    <RenderDropdownList
      listItem={
        Array [
          "Sport",
          "Entertainment",
          "Economics",
          "Technology",
          "Games",
        ]
      }
      titleList="Section"
    />
    <a
      href="#editorial"
    >
      Editorial
    </a>
    <a
      href="#contactus"
    >
      Contact Us
    </a>
  </Nav>
  <Nav
    displayName="Nav"
  >
    <a
      href="#avatar"
    >
      <Avatar
        displayName="Avatar"
        size="xs"
        title="JS"
      />
    </a>
  </Nav>
</Navbar>
`;
